### PR TITLE
Release 2.5.0

### PR DESCRIPTION
**Pull request by d4straub**
Summary of changes:
- Added COIDB DADA2 taxonomic database
- Small adjustments to taxonomic classification, ASV filtering by barrnap, fasta input
- sbdi-export works for more taxonomic databases now
- nf-core template update from 2.6 to 2.7.2

See CHANGELOG for detailed changes.

I started the AWS full test dataset on branch dev. Edit: AWS full test was successful!

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by erikrikarddaniel — APPROVED** [review]
Almost impossible to check so many files, but since we've been through all individual PRs: LGTM.

**Comment by Emiller88** [comment]
Just wondering about the release number. I think this should be a minor version bump, not a patch release.

>MINOR version when you add functionality in a backwards compatible manner

https://semver.org/

**Comment by d4straub** [comment]
@Emiller88 yes, there comes (minor) additional functionality, or rather adjustments, with the release, so I choose the patch release. But I agree, a minor release, i.e. 2.5.0, would be also a possible choice I think.
Would you be willing to make a full review here? That would be really needed :)

**Comment by Emiller88** [comment]
@d4straub Sure I'd be happy to review!

> Given a version number MAJOR.MINOR.PATCH, increment the:
    MAJOR version when you make incompatible API changes
    MINOR version when you add functionality in a backwards compatible manner
    PATCH version when you make backwards compatible bug fixes
Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

You have things in the `ADDED` section of the CHANGELOG which warrants a minor version, not a patch version. I don't see semantic versioning as a “choice” of a version, that's the idea is to remove the “choice” and just use a set of rules to dictate the release number increment.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,41 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## nf-core/ampliseq version 2.4.2 - 2023-03-01
```
Comment (Emiller88) [inline CHANGELOG.md:6]:
```suggestion
## nf-core/ampliseq version 2.5.0 - 2023-03-01
```
After reviewing, this has a ton of added functionality. Which is a good thing, it's just not a patch in my understanding of semantic versioning.

File: nextflow.config [hunk 2]
```diff
@@ -250,12 +255,12 @@ dag {
...
-    author          = 'Daniel Straub, Alexander Peltzer'
+    author          = """Daniel Straub, Alexander Peltzer"""
     homePage        = 'https://github.com/nf-core/ampliseq'
-    description     = 'Amplicon sequencing analysis workflow using DADA2 and QIIME2'
+    description     = """Amplicon sequencing analysis workflow using DADA2 and QIIME2"""
     mainScript      = 'main.nf'
-    nextflowVersion = '!>=21.10.3'
-    version         = '2.4.1'
+    nextflowVersion = '!>=22.10.1'
+    version         = '2.4.2'
```
Comment (Emiller88) [inline nextflow.config:263]:
```suggestion
    version         = '2.5.0'
```

File: workflows/ampliseq.nf [hunk 3]
```diff
@@ -604,7 +617,7 @@ workflow AMPLISEQ {
     if ( params.sbdiexport ) {
         SBDIEXPORT ( ch_dada2_asv, ch_dada2_tax, ch_metadata )
         ch_versions = ch_versions.mix(SBDIEXPORT.out.versions.first())
-        SBDIEXPORTREANNOTATE ( ch_dada2_tax )
+	    SBDIEXPORTREANNOTATE ( ch_dada2_tax, ch_barrnapsummary )
```
Comment (Emiller88) [inline workflows/ampliseq.nf:620]:
```suggestion
        SBDIEXPORTREANNOTATE ( ch_dada2_tax, ch_barrnapsummary )
```
Random tab

**Review by Emiller88 — CHANGES_REQUESTED** [review]
Happy to get a second opinion on this, but I think it should be a minor version release. Other than that it looks great!

File: CHANGELOG.md (hunk 1 above)
Comment (d4straub) [reply CHANGELOG.md:6]: Alright, I am fine with naming it 2.5.0, did so in https://github.com/nf-core/ampliseq/pull/550

File: nextflow.config (hunk 2 above)
Comment (d4straub) [reply nextflow.config:263]: done in https://github.com/nf-core/ampliseq/pull/550

File: workflows/ampliseq.nf (hunk 3 above)
Comment (d4straub) [reply workflows/ampliseq.nf:620]: Good find! Also fixed in https://github.com/nf-core/ampliseq/pull/550

**Review by Emiller88 — APPROVED** [review]

**Pull request merged**